test/index.js

```diff
@@ -19,16 +31,28 @@ describe('switchPath', () => {
         '/foo': 456,
       },
     });
+
     expect(path).to.be.equal('/home/foo');
     expect(value).to.be.equal(456);
   });
 
-  it('should match a path on an incomplete pattern', () => {
+  it('should NOT match a path on an incomplete pattern', () => {
     const {path, value} = switchPath('/home/foo', {
       '/bar': 123,
       '/home': 456,
     });
-    expect(path).to.be.equal('/home');
+    expect(path).to.be.equal(null);
+    expect(value).to.be.equal(null);
+  });
```

Comment:
I really can't accept this test, this is the opposite direction of what I want switchPath to accomplish. Optimistically matching a path based on a prefix is super important for how I want to use it for nested dialogues in Cycle. The case being: imagine the parent using switchPath on
```
const {path, Component} = switchPath(fullPath, {
  '/about': AboutComponent,
  '/profile': ProfileComponent,
  '/movies': MoviesComponent,
}
```
SwitchPath returns the right hand side as `Component`, which is a dialogue function.
Then the parent calls `Component` and passes to it both the `fullPath` and `path`. Then the Component (for instance `MoviesComponent` removes `path` as a prefix of `fullPath` to get `subPath`, and internally the child Component will do another switchPath on that subPath:
```js
const {path, vtree} = switchPath(subPath, {
  '/': moviesHomeVTree,
  '/:id': movieDetailVTree,
})
```
See the importance of that feature now?